tests/test_snql_api.py

```diff
@@ -255,3 +275,32 @@ def test_project_rate_limiting_subqueries(self) -> None:
             ),
         )
         assert response.status_code == 429
+
+    @patch("snuba.settings.RECORD_QUERIES", True)
+    @patch("snuba.state.record_query")
+    def test_record_queries(self, record_query_mock: Any) -> None:
```

Comment:
Did this test get split up into 2 separate files now? How come it doesn't use the `_build_snql_post_methods` magic?

Reply:
I duplicated this test because the expected_call count of the MagicMock gets messed up by the pytest params. pytest also has a bug that doesn't allow me to see which param the test is being run as (legacy/snql/compare) so I can't even do something like `expected = 1 if param == "legacy" else 2` or whatever. It was easier to have two tests (that don't use params) to test the Legacy and the SnQL side of things.

Reply:
Do you know what causes the call count to be different in SnQL though? Given a same query just in SnQL format shouldn't the splitter result in the same number of underlying ClickHouse queries to be run?

Reply:
Sorry, the test checks how many times the `record_query_mock` has been called. If you run the test more than once with params, the number of calls becomes cumulative. So when the legacy versions runs the expected call count is 1, but in the compare version it ends up at 4 I think.